components/server/src/config.ts

```diff
@@ -0,0 +1,203 @@
+/**
+ * Copyright (c) 2020 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import { GitpodHostUrl } from '@gitpod/gitpod-protocol/lib/util/gitpod-host-url';
+import { AuthProviderParams } from './auth/auth-provider';
+
+import { Branding, NamedWorkspaceFeatureFlag } from '@gitpod/gitpod-protocol';
+
+import { RateLimiterConfig } from './auth/rate-limiter';
+import { Env } from './env';
+import { CodeSyncConfig } from './code-sync/code-sync-service';
+
+export const Config = Symbol("Config");
+export interface Config {
+    version: string;
+    hostUrl: GitpodHostUrl;
+
+    license: string | undefined;
+    trialLicensePrivateKey: string | undefined;
+
+    heartbeat: {
+        intervalSeconds: number;
+        timeoutSeconds: number,
+    };
+
+    workspaceDefaults: {
+        ideVersion: string;
+        ideImageRepo: string;
+        ideImage: string;
+        ideImageAliases: { [index: string]: string };
+        workspaceImage: string;
+        previewFeatureFlags: NamedWorkspaceFeatureFlag[];
+        defaultFeatureFlags: NamedWorkspaceFeatureFlag[];
+    };
+
+    session: {
+        maxAgeMs: number;
+        secret: string;
+    };
+
+    // TODO(gpl): app.githubHost / GHE_HOST relevant ???
+    githubApp: {
+        enabled: boolean;
+        appId: number;
+        webhookSecret: string;
+        authProviderId: string;
+        certPath: string;
+        marketplaceName: string;
+        logLevel?: string;
+    };
+
+    definitelyGpDisabled: boolean;
+
+    workspaceGarbageCollection: {
+        disabled: boolean;
+        startDate: number;
+        chunkLimit: number;
+        minAgeDays: number;
+        minAgePrebuildDays: number;
+        contentRetentionPeriodDays: number;
+        contentChunkLimit: number;
+    };
+
+    enableLocalApp: boolean;
+
+    authProviderConfigs: AuthProviderParams[];
+    disableDynamicAuthProviderLogin: boolean;
+
+    // TODO(gpl) Can we remove this?
+    brandingConfig: Branding;
```

Comment:
I guess so. 
The single usage I've seen recently was the sanity check for returnTo URLs where we want to allow redirects to the configured `homepage`, which in turn can be remodeled as config attribute in here.